libmscore/score.cpp

```diff
@@ -2067,59 +2034,132 @@ bool Score::appendScore(Score* score, bool addPageBreak, bool addSectionBreak)
+                  KeySigEvent nkse = KeySigEvent(ostaff->keySigEvent(otick));
+                  KeySig* nks = new KeySig(this);
+                  nks->setScore(this);
+                  nks->setTrack(trackIdx);
+
+                  nks->setKeySigEvent(nkse);
+                  staff->setKey(ctick, nkse);
+                  ns->add(nks);
+                  //undoAddElement(nks);
+                  }
+            // check if a key signature is present but is spurious (i.e. no actual change)
+            else if (staff->currentKeyTick(ctick) == ctick &&
+                     staff->key(ctick - Fraction::fromTicks(1)) == ostaff->key(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::KeySig);
+                        if (ns) ns->remove(ns->element(trackIdx));
+                  }
+            
+            // check if time signature needs to be changed
+            if (!ostaff->timeSig(otick) || !staff->timeSig(ctick)) {} // yes, this can happen.
+            else if (!((*ostaff->timeSig(otick)) == (*staff->timeSig(ctick)))) {
+                  Segment* ns = firstAppendedMeasure->undoGetSegment(SegmentType::TimeSig, ctick);
+                  TimeSig* nsig = new TimeSig(*(ostaff->timeSig(otick)));
+
+                  nsig->setScore(this);
+                  nsig->setTrack(trackIdx);
+                  ns->add(nsig);
+                  //undoAddElement(nsig);
                   }
+            // check if a time signature is present but is spurious (i.e. no actual change)
+            else if (staff->currentTimeSigTick(ctick) == ctick &&
+                     (*staff->timeSig(ctick - Fraction::fromTicks(1))) == (*ostaff->timeSig(otick))) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::TimeSig);
+                        if (ns) ns->remove(ns->element(trackIdx));
+                  }
+
+            // check if clef signature needs to be changed
+            if (!(ostaff->clef(otick) == staff->clef(ctick))) {
+                  undoChangeClef(staff,firstAppendedMeasure,ostaff->clef(otick));
+                  }
+            // check if a clef change is present but is spurious (i.e. no actual change)
+            else if (staff->currentClefTick(ctick) == ctick &&
+                     staff->clef(ctick - Fraction::fromTicks(1)) == ostaff->clef(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::Clef);
+                        if (!ns) ns = firstAppendedMeasure->first(SegmentType::HeaderClef);
+                        if (ns) ns->remove(ns->element(trackIdx));
+                  }
+            }
+
+      // check if section starts with apick-up measure to be merged with end of previous section
+      Measure* cm = firstAppendedMeasure, * pm = cm->prevMeasure();
+      if (pm->timesig() == cm->timesig() && 
+          pm->ticks() + cm->ticks() == cm->timesig()) {
```

Comment:
Do you really need to separate this into two lines? You use equally long if statements in other spots. Feel free to explain your reasoning.